rts/Lua/LuaOpenGL.cpp

```diff
@@ -4147,6 +4154,52 @@ int LuaOpenGL::Clear(lua_State* L)
 	return 0;
 }
 
+namespace Impl {
+	template<class Type, auto glClearBufferFuncPtrPtr>
+	static inline void ClearBuffer(lua_State* L, GLenum bufferType, GLint drawBuffer) {
+		Type values[4];
+		values[0] = luaL_optnumber(L, 3, 0);
+		values[1] = luaL_optnumber(L, 4, 0);
+		values[2] = luaL_optnumber(L, 5, 0);
+		values[3] = luaL_optnumber(L, 6, 0);
+		(*glClearBufferFuncPtrPtr)(bufferType, drawBuffer, values);
+	}
+}
+
+int LuaOpenGL::ClearBuffer(lua_State* L)
+{
```

Comment:
I dislike 0, -1 and nils in this API, as much as possible the API should be self-explainatory, which is not the case here. Use semantics of https://springrts.com/wiki/Lua_FBO_and_RBO ("depth", "stencil", "color0", etc) or literal GL constants.

Also there's no need to for a user to explicitly select what kind of format the attached texture has, the implementation should either store it somewhere or retrieve it from the GL server - https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGetFramebufferAttachmentParameter.xhtml

Reply:
> I dislike 0, -1 and nils in this API, as much as possible the API should be self-explainatory, which is not the case here. Use semantics of https://springrts.com/wiki/Lua_FBO_and_RBO ("depth", "stencil", "color0", etc) or literal GL constants.
> 
> Also there's no need to for a user to explicitly select what kind of format the attached texture has, the implementation should either store it somewhere or retrieve it from the GL server - https://registry.khronos.org/OpenGL-Refpages/gl4/html/glGetFramebufferAttachmentParameter.xhtml

Why don't we keep attachments as part of FBO class ourselves?

Reply:
Because it was rarely (if it all needed) before

Reply:
Going to update this later.

Reply:
There is good possibility thing glGetFramebuffer... is busted.
I may be needing to update FBO class to retain info of its attachments.

Reply:
What do you mean "busted"?

Reply:
> What do you mean "busted"?

Not busted, I got it working in the end.
But this function is limited. The only useful is GL_FRAMEBUFFER_ATTACHMENT_COMPONENT_TYPE, but that is not the only thing I need.
I am planning to make Lua FBOs store attachment formats tomorrow, so they can be retrieved. What do you think?

Reply:
Yeah no reason not to store additional information, if you need it.